src/main/java/org/opensearch/knn/quantization/models/quantizationOutput/BinaryQuantizationOutput.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.models.quantizationOutput;
+
+/**
+ * The BinaryQuantizationOutput class represents the output of a quantization process in binary format.
+ * It implements the QuantizationOutput interface to handle byte arrays specifically.
+ */
+public class BinaryQuantizationOutput implements QuantizationOutput<byte[]> {
```

Comment:
I want to see if we can avoid wrapping the array with and object. @navneet1v had tried something similar with KNNVector and an increase in memory usage was seen with the approach. Maybe he can share some insights.

Can you throw some light on how this is intended to be used when integrated with say indexing. Will we be able to get away with simply returning byte array without this wrapper in quantizer?

Reply:
Un-resolving the comment as the comment is not resolved to say what approach we are going to follow here.

Reply:
Sure , we are following the approach where we will pass QuantizationOutput in quantize Method , which avoids creating new object for everryQuantization and I had implemented quantized output in such as way that even the underlying quantized vector input reference will also be same .